### ISSUE TITLE
libtiff source no longer available (new patch level)

### ISSUE THREAD

**Issue by tseemann** [Homebrew/homebrew-core#3264]
```
==> Downloading http://download.osgeo.org/libtiff/tiff-4.0.6.tar.gz
Already downloaded: /home/tseemann/.cache/Homebrew/libtiff-4.0.6.tar.gz
==> Downloading https://mirrors.ocf.berkeley.edu/debian/pool/main/t/tiff/tiff_4.0.6-1.debian.tar.xz

curl: (22) The requested URL returned error: 404 Not Found
Trying a mirror...
==> Downloading https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/t/tiff/tiff_4.0.6-1.debian.tar.xz

curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "libtiff--patch"
Download failed: https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/t/tiff/tiff_4.0.6-1.debian.tar.xz
```

Seems to be `-2` patch level now and old .tar.xz is gone :-(

https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/t/tiff/tiff_4.0.6-2.debian.tar.xz

**Comment by DomT4** [comment]
Fixed via https://github.com/Homebrew/homebrew-core/commit/d58f19cfe052ee357ad6885d62ee7fca8c264cf3 & https://github.com/Homebrew/homebrew-core/commit/c8f9c5681adb4f18e415adeb59b3841e501b8861. Thanks for the report!

**Comment by ope-sal** [comment]
has this been fixed. Running into the same issue

**Comment by DomT4** [comment]
Yes. You may well need to `brew update`.

**Comment by ope-sal** [comment]
worked, thanks!

**Comment by urbanophile** [comment]
I have `brew update`'d and I still receive the same error. 
`$ brew install wine
==> Installing dependencies for wine: libtiff, webp, gd, libgphoto2, little-cms2, jasper, libicn
==> Installing wine dependency: libtiff
==> Downloading http://download.osgeo.org/libtiff/tiff-4.0.6.tar.gz
Already downloaded: /Users/matt/Library/Caches/Homebrew/libtiff-4.0.6.tar.gz
==> Downloading https://mirrors.ocf.berkeley.edu/debian/pool/main/t/tiff/tiff_4.0.6-1.debian.tar.xz

curl: (22) The requested URL returned error: 404 Not Found
Trying a mirror...
==> Downloading https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/t/tiff/tiff_4.0.6-1.deb

curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "libtiff--patch"
Download failed: https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/t/tiff/tiff_4.0.6-1.debian.tar.xz
$ brew update
Already up-to-date.
`

**Comment by rpmonteiro** [comment]
Same problem as @urbanophile

**Comment by DomT4** [comment]
Some fairly strange errors there. For a start, you've apparently got a mirror pointing to a `.deb` which I can't imagine would have ever been the case in Homebrew, for obvious enough reasons.

This is what you should be seeing:
```
==> Downloading http://download.osgeo.org/libtiff/tiff-4.0.6.tar.gz
/usr/bin/curl --fail --remote-time --location --user-agent Homebrew/0.9.9 (Macintosh; Intel Mac OS X 10.11.6) curl/7.43.0 http://download.osgeo.org/libtiff/tiff-4.0.6.tar.gz -C 0 -o /usr/local/var/homebrew/cache/libtiff-4.0.6.tar.gz.incomplete --connect-timeout 5
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2141k  100 2141k    0     0   416k      0  0:00:05  0:00:05 --:--:--  416k
Downloaded to: /usr/local/var/homebrew/cache/libtiff-4.0.6.tar.gz
SHA256: 4d57a50907b510e3049a4bba0d7888930fdfc16ce49f1bf693e5b6247370d68c
==> Verifying libtiff-4.0.6.tar.gz checksum
==> Downloading https://mirrors.ocf.berkeley.edu/debian/pool/main/t/tiff/tiff_4.0.6-2.debian.tar.xz
/usr/bin/curl --fail --remote-time --location --user-agent Homebrew/0.9.9 (Macintosh; Intel Mac OS X 10.11.6) curl/7.43.0 https://mirrors.ocf.berkeley.edu/debian/pool/main/t/tiff/tiff_4.0.6-2.debian.tar.xz -C 0 -o /usr/local/var/homebrew/cache/libtiff--patch-82a0ef3f713d2a22d40b9be71fd121b9136657d313ae6b76b51430302a7b9f8b.tar.xz.incomplete --connect-timeout 5
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 19960  100 19960    0     0  15312      0  0:00:01  0:00:01 --:--:-- 15318
Downloaded to: /usr/local/var/homebrew/cache/libtiff--patch-82a0ef3f713d2a22d40b9be71fd121b9136657d313ae6b76b51430302a7b9f8b.tar.xz
SHA256: 82a0ef3f713d2a22d40b9be71fd121b9136657d313ae6b76b51430302a7b9f8b
```

Try:
```bash
cd $(brew --prefix) && git fetch && git reset --hard origin/master && cd $(brew --prefix)/Library/Taps/homebrew/homebrew-core && git fetch && git reset --hard origin/master
```

**Comment by zvictor** [comment]
I was having the same error, then I ran the following commands

```haskell
$ brew update
$ brew cleanup
$ rm -Rf ~/Library/Caches/Homebrew/
```

but now I get `Error: Empty archive`.

```haskell
$ brew install wine
Warning: You are using a pre-release version of Xcode.
You may encounter build failures or other breakages.
Please create pull-requests instead of filing issues.
==> Installing dependencies for wine: libtool, fontconfig, libtiff, webp, gd, libgphoto2, little-cms2, jasper, libicns, makedepend, openssl,
==> Installing wine dependency: libtool
==> Downloading https://ftpmirror.gnu.org/libtool/libtool-2.4.6.tar.xz
==> Downloading from http://ftp.download-by.net/gnu/gnu/libtool/libtool-2.4.6.tar.xz
######################################################################## 100.0%
Error: Empty archive
```

**Comment by urbanophile** [comment]
@DomT4 Thanks for your help, doing the git hard reset to origin/master fixed the problem. I am not sure why there was so many weird errors with my configuration of homebrew.

**Comment by mgoofyy** [comment]
Same problem as @rpmonteiro

**Comment by DomT4** [comment]
Please try `brew update`. If that doesn't work, try the:
```bash
cd $(brew --prefix) && git fetch && git reset --hard origin/master && cd $(brew --prefix)/Library/Taps/homebrew/homebrew-core && git fetch && git reset --hard origin/master
```
step. There's no new Debian patch blob upstream, so the URL is still good.

**Comment by stephane-lb** [comment]
Hi,

Sorry to re-open this one but I did that : 
```
cd $(brew --prefix) && git fetch && git reset --hard origin/master && cd $(brew --prefix)/Library/Taps/homebrew/homebrew-core && git fetch && git reset --hard origin/master && brew update 
```

and i just got this error : 
```                                                                                                                                                                              
Stephane-lb@Arcadia /usr/local                                                                                                                                            [18:49:48] 
> $ brew install openimageio                                                                                                                                        [±master ✓]
==> Installing openimageio from homebrew/science
==> Installing dependencies for homebrew/science/openimageio: libtiff, little-cms2, opencolorio, openjpeg, webp, opencv
==> Installing homebrew/science/openimageio dependency: libtiff
==> Using the sandbox
==> Downloading http://download.osgeo.org/libtiff/tiff-4.0.6.tar.gz
######################################################################## 100.0%
==> Downloading https://mirrors.ocf.berkeley.edu/debian/pool/main/t/tiff/tiff_4.0.6-2.debian.tar.xz
######################################################################## 100.0%
==> Patching
==> Applying patches/01-CVE-2015-8665_and_CVE-2015-8683.patch
patching file ChangeLog
patching file libtiff/tif_getimage.c
==> Applying patches/02-fix_potential_out-of-bound_writes_in_decode_functions.patch
patching file ChangeLog
patching file libtiff/tif_luv.c
==> Applying patches/03-fix_potential_out-of-bound_write_in_NeXTDecode.patch
patching file ChangeLog
patching file libtiff/tif_next.c
==> Applying patches/04-CVE-2016-5314_CVE-2016-5316_CVE-2016-5320_CVE-2016-5875.patch
patching file ChangeLog
patching file libtiff/tif_pixarlog.c
==> Applying patches/05-CVE-2016-6223.patch
patching file ChangeLog
patching file libtiff/tif_read.c
==> Applying patches/06-CVE-2016-5321.patch
patching file ChangeLog
patching file tools/tiffcrop.c
==> Applying patches/07-CVE-2016-5323.patch
patching file ChangeLog
patching file tools/tiffcrop.c
==> ./configure --prefix=/usr/local/Cellar/libtiff/4.0.6_2 --without-x --with-jpeg-include-dir=/usr/local/opt/jpeg/include --with-jpeg-lib-dir=/usr/local/opt/jpeg/lib --disable
==> make install
Last 15 lines from /Users/Stephane-lb/Library/Logs/Homebrew/libtiff/02.make:
      ___clang_call_terminate in tif_stream.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& std::__1::__put_character_sequence<char, std::__1::char_traits<char> >(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, char const*, unsigned long) in tif_stream.o
  "___cxa_end_catch", referenced from:
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& std::__1::__put_character_sequence<char, std::__1::char_traits<char> >(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, char const*, unsigned long) in tif_stream.o
  "___gxx_personality_v0", referenced from:
      TIFFStreamOpen(char const*, std::__1::basic_ostream<char, std::__1::char_traits<char> >*) in tif_stream.o
      _tiffosSeekProc(void*, unsigned long, int) in tif_stream.o
      _tiffosSizeProc(void*) in tif_stream.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& std::__1::__put_character_sequence<char, std::__1::char_traits<char> >(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, char const*, unsigned long) in tif_stream.o
      std::__1::ostreambuf_iterator<char, std::__1::char_traits<char> > std::__1::__pad_and_output<char, std::__1::char_traits<char> >(std::__1::ostreambuf_iterator<char, std::__1::char_traits<char> >, char const*, char const*, char const*, std::__1::ios_base&, char) in tif_stream.o
      Dwarf Exception Unwind Info (__eh_frame) in tif_stream.o
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [libtiffxx.la] Error 1
make: *** [install-recursive] Error 1
```

Do you have any idea of what it could be ? 

Thanks for your precious help,

Stephane-lb

**Comment by DomT4** [comment]
Interesting error. Might be worth opening a new issue for that, but please `brew gist-logs libtiff` as that contains a bunch of build information we'll need to look at it properly.

**Comment by stephane-lb** [comment]
Ok, let's do that. 

gist log available here : https://gist.github.com/stephane-lb/6ecf18d2eff15b65051ae2107083a684

**Comment by DomT4** [comment]
@stephane-lb Try `brew unlink llvm` and then re-do the build.

**Comment by stephane-lb** [comment]
Thanks @DomT4 : it works!

**Comment by DomT4** [comment]
Great! 👍. Sadly `llvm` is one of the more problematic ones to force link.

**Comment by bennyscetbun** [comment]
```shell
$> cd $(brew --prefix) && git fetch && git reset --hard origin/master && cd $(brew --prefix)/Library/Taps/homebrew/homebrew-core && git fetch && git reset --hard origin/master
HEAD is now at 54884cd Merge pull request #903 from zmwangx/always-show-dev-commands
remote: Counting objects: 20, done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 20 (delta 15), reused 18 (delta 13), pack-reused 0
Unpacking objects: 100% (20/20), done.
From https://github.com/Homebrew/homebrew-core
   93724fe..2f2000f  master     -> origin/master
HEAD is now at 2f2000f docker-machine: update 0.8.1 bottle.
$> brew cleanup
Warning: Skipping go: most recent version 1.7.1 not installed
Warning: Skipping node: most recent version 6.5.0 not installed
Warning: Skipping protobuf: most recent version 3.0.2 not installed
Warning: Skipping sqlite: most recent version 3.14.2 not installed
Warning: Skipping watch: most recent version 3.3.12 not installed
$> brew update
Already up-to-date.
$> rm -Rf ~/Library/Caches/Homebrew/
$> brew install wine winetricks
==> Installing dependencies for wine: libtiff, webp, gd, libgphoto2, little-cms2, jasper, libicns, makedepend, openssl, net-snmp, sane-backends, libtasn1, gmp, nettle, gnutls
==> Installing wine dependency: libtiff
==> Using the sandbox
==> Downloading http://download.osgeo.org/libtiff/tiff-4.0.6.tar.gz
######################################################################## 100.0%
==> Downloading https://mirrors.ocf.berkeley.edu/debian/pool/main/t/tiff/tiff_4.0.6-2.debian.tar.xz
######################################################################## 100.0%
==> Patching
Error: Empty archive
```
What am i doing wrong?

**Comment by DomT4** [comment]
@bennyscetbun Try `brew update && brew reinstall xz`.

**Comment by bennyscetbun** [comment]
Victory :)
So the complete solution would be:
```
cd $(brew --prefix) && git fetch && git reset --hard origin/master && cd $(brew --prefix)/Library/Taps/homebrew/homebrew-core && git fetch && git reset --hard origin/master
brew update
brew cleanup
rm -Rf ~/Library/Caches/Homebrew/
brew reinstall xz
brew install wine
```

Thanks a lot @DomT4

**Comment by fuchao2012** [comment]
@bennyscetbun thanks for the clearly instructions, but `brew update` runs soooo slow, why is that?

**Comment by MikeMcQuaid** [comment]
@fuchao2012 There's a few one-time migrations that may happen on `brew update`.

**Comment by Jolg42** [comment]
You can replace 
`cd $(brew --prefix) && git fetch && git reset --hard origin/master && cd $(brew --prefix)/Library/Taps/homebrew/homebrew-core && git fetch && git reset --hard origin/master`

by this new one for recent Homebrew versions because the paths changed!

`cd $(brew --prefix)/Homebrew && git fetch && git reset --hard origin/master && cd $(brew --prefix)/Homebrew/Library/Taps/homebrew/homebrew-core && git fetch && git reset --hard origin/master`

**Comment by MikeMcQuaid** [comment]
@Jolg42 Instead of `$(brew --prefix)/Homebrew` use `$(brew --repo)`

**Comment by ilovezfs** [comment]
also
```
$(brew --prefix)/Homebrew/Library/Taps/homebrew/homebrew-core
```
can be
```
$(brew --repo homebrew/core)
```

**Comment by CodeMaxx** [comment]
I'm also having this issue. `brew` tries to download v 4.0.7-4 while the one available on the mirror is 4.0.7-5

**Comment by ilovezfs** [comment]
This will be addressed by https://github.com/Homebrew/homebrew-core/pull/9409.
